src/aleph/handlers/content/vm.py

```diff
@@ -320,15 +320,19 @@ class VmMessageHandler(ContentHandler):
     """
 
     async def check_balance(self, session: DbSession, message: MessageDb) -> None:
-        if message.type != MessageType.instance:
+        if not (message.type == MessageType.instance) and not (message.type == MessageType.program):
```

Comment:
`_get_vm_content()` called below already performs this check, you can just remove this line.